database/migrations/2025_12_28_000000_update_comment_table.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+declare(strict_types=1);
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration {
+    public function up(): void
+    {
+        // Delete orphaned comments with unmapped ArticleType values.
+        // (News=4, Forum=8, invalid=0, or any other unexpected values)
+        DB::statement("DELETE FROM Comment WHERE ArticleType NOT IN (1, 2, 3, 5, 6, 7, 9, 10, 11, 12)");
+
+        // Populate commentable_type from ArticleType.
+        DB::statement("UPDATE Comment SET commentable_type = CASE ArticleType
+            WHEN 1 THEN 'game.comment'
+            WHEN 2 THEN 'achievement.comment'
+            WHEN 3 THEN 'user.comment'
+            WHEN 5 THEN 'user-activity.comment'
+            WHEN 6 THEN 'leaderboard.comment'
+            WHEN 7 THEN 'trigger.ticket.comment'
+            WHEN 9 THEN 'user-moderation.comment'
+            WHEN 10 THEN 'game-hash.comment'
+            WHEN 11 THEN 'achievement-set-claim.comment'
+            WHEN 12 THEN 'game-modification.comment'
+        END WHERE commentable_type IS NULL");
+
+        // Drop the index first, then the column (required for SQLite compatibility).
+        Schema::table('Comment', function (Blueprint $table) {
+            $table->dropIndex('comments_commentable_index');
+        });
+        Schema::table('Comment', function (Blueprint $table) {
+            $table->dropColumn('commentable_id');
+        });
+
+        // Drop indexes before renaming columns, otherwise the migration may fail.
+        $driver = Schema::getConnection()->getDriverName();
+        if ($driver !== 'sqlite') {
+            DB::statement('SET FOREIGN_KEY_CHECKS=0');
+        }
+        Schema::table('Comment', function (Blueprint $table) {
+            $table->dropIndex('comment_articleid_index');
+            $table->dropIndex('comment_user_id_submitted_index');
+        });
+        if ($driver !== 'sqlite') {
+            DB::statement('SET FOREIGN_KEY_CHECKS=1');
+        }
+
+        Schema::table('Comment', function (Blueprint $table) {
+            $table->renameColumn('ArticleID', 'commentable_id');
```

Comment:
Because the `commentable_type` column is kept and this column is renamed, `commentable_type` appears out-of-order:
```
> select * from comments where id=12345\G
*************************** 1. row ***************************
              id: 12345
  commentable_id: 5017
         user_id: 11371
commentable_type: achievement.comment
            body: It's somewhat RNG-dependent.  He got really good spawns in the vertical shaft.  If you get the wrong ones, it's not happening no matter how good you are.
      created_at: 2015-09-14 20:36:41
      updated_at: NULL
      deleted_at: NULL
1 row in set (0.002 sec)
```

It should be before `commentable_id`.

Reply:
This should be fixed in the latest commit.